json_schemas/buyer-email-domains.json

```diff
@@ -0,0 +1,14 @@
+{
+  "$schema": "http://json-schema.org/schema#",
+  "additionalProperties": false,
+  "properties": {
+    "domainName": {
+      "pattern": "^(?!\\.)([^:\/@!&=\\s])+\\.[a-zA-Z]{2,64}?$",
```

Comment:
This looks like it will work as far as I can tell (and we have tests to reassure us).  

I'm not sure we need so many brackets in it, and I'm not sure the nonmatching lookahead group at the start adds anything more than a straight "not a dot" to start?

So I think this does the same thing, perhaps marginally more readable, but it is marginal:
```
^[^\.][^:/@!&=\s]+\.[a-zA-Z]{2,64}?$
```
(Probably with added escapes for the slashes)

Reply:
And I don't think we need the final `?` - if it's matching the start and end of the line then greedy/non-greedy makes no sense - it's going to match it all whatever.
So I think `^[^\.][^:/@!&=\s]+\.[a-zA-Z]{2,64}$` is equivalent.

Reply:
Aha!  OK so your negative capturing group means yours will match single-character before the domain e.g. `a.com`, but mine requires at least two characters before the dot (so `a.com` will fail but `ab.com` will match).

Which of these we want, I'm not 100% sure... probably stick with yours?

Reply:
You're right - I'm used to grouping the different bits to match and use later, but we don't need that here. Tests pass without the lookahead and the `?`, pushing a commit now 😸

Reply:
I'm not keen on `a.com`! They aren't going to come up very much for government buyers: https://en.wikipedia.org/wiki/Single-letter_second-level_domain

Reply:
Though I suppose something like `a.cymru` might. Hmm, will fiddle a bit and see what I can do.

Reply:
Have reinstated the lookahead but simplified the rest of the regex.